File: tests/fixtures/hello_world.c
/*
 * hello_world.c – simple C fixture used by AXE feature-validation tests.
 *
 * Purpose: provide a small, self-contained C source file that AXE agents can
 * read, analyse, and comment on during automated testing of batch mode, collab
 * mode, and the /prep / /llmprep / /buildinfo interactive commands.
 *
 * Build:
 *   gcc -o hello_world hello_world.c
 */
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

/* Structure representing a simple key-value pair. */
typedef struct {
    char key[32];
    int  value;
} Entry;

/* Populate an array of entries. */
static void populate(Entry *entries, int n) {
    int i;
    for (i = 0; i < n; i++) {
        snprintf(entries[i].key, sizeof(entries[i].key), "item_%d", i);
        entries[i].value = i * i;
    }
}

/* Print the entry array to stdout. */
static void print_entries(const Entry *entries, int n) {
    int i;
    for (i = 0; i < n; i++) {
        printf("%s = %d\n", entries[i].key, entries[i].value);
    }
}

int main(void) {
    const int N = 5;
    Entry *entries = (Entry *)malloc(N * sizeof(Entry));
    if (!entries) {
        fprintf(stderr, "malloc failed\n");
        return 1;
    }

    populate(entries, N);
    print_entries(entries, N);

    free(entries);
    return 0;
}

File: tests/fixtures/hello_world.py
#!/usr/bin/env python3
"""
hello_world.py – simple Python fixture used by AXE feature-validation tests.

Purpose: provide a small, well-structured Python module that AXE agents can
read, analyse, and review during automated testing.  Covers type hints,
docstrings, dataclasses, and simple I/O – features agents are expected to
understand and comment on correctly.
"""
from __future__ import annotations

from dataclasses import dataclass, field
from typing import List


@dataclass
class Entry:
    """A simple key/value entry."""

    key: str
    value: int


def populate(n: int) -> List[Entry]:
    """
    Create a list of *n* entries with sequential squared values.

    Args:
        n: Number of entries to create.

    Returns:
        List of Entry objects.
    """
    return [Entry(key=f"item_{i}", value=i * i) for i in range(n)]


def print_entries(entries: List[Entry]) -> None:
    """Print each entry to stdout."""
    for entry in entries:
        print(f"{entry.key} = {entry.value}")


def main() -> None:
    entries = populate(5)
    print_entries(entries)


if __name__ == "__main__":
    main()
